frontend/src/entry/model/mirroredModels/ReleasesSelector.tsx

```diff
@@ -15,49 +15,48 @@ type ReleasesSelectorProps = {
 
 export default function ReleaseSelector({ model, selectedReleases, setSelectedReleases }: ReleasesSelectorProps) {
   const [checked, setChecked] = useState(false)
-  const [errorMessage, setErrorMessage] = useState('')
   const { releases, isReleasesLoading, isReleasesError } = useGetReleasesForModelId(model.id)
-  if (isReleasesLoading) {
-    return <Loading />
-  }
-  if (isReleasesError) {
-    setErrorMessage(isReleasesError.info.message)
-  }
 
-  const handleRemoveRelease = async (removedRelease: ReleaseInterface) => {
-    setChecked(false)
-    setSelectedReleases(selectedReleases.filter((release) => release.semver !== removedRelease.semver))
-  }
+  const selectedReleasesDisplay = useMemo(() => {
+    const handleRemoveRelease = async (removedRelease: ReleaseInterface) => {
+      setChecked(false)
+      setSelectedReleases(selectedReleases.filter((release) => release.semver !== removedRelease.semver))
+    }
+    return selectedReleases.map((release) => (
+      <>
+        <Grid item xs={10}>
+          <ReleaseDisplay key={release.semver} model={model} release={release} hideReviewBanner />
+        </Grid>
+        <Grid item xs={2}>
+          <Tooltip title={'Remove'}>
+            <IconButton onClick={() => handleRemoveRelease(release)}>
+              <DeleteIcon color={'error'} />
+            </IconButton>
+          </Tooltip>
+        </Grid>
+      </>
+    ))
+  }, [selectedReleases, model, setSelectedReleases])
 
   const handleChecked = async (event: ChangeEvent<HTMLInputElement>) => {
     setSelectedReleases(event.target.checked ? releases : [])
     setChecked(event.target.checked)
   }
 
+  if (isReleasesLoading) {
+    return <Loading />
+  }
+  if (isReleasesError) {
+    ;<MessageAlert message={isReleasesError.info.message} severity='error' />
```

Comment:
My bad for giving you a broken code snippet. This should return the MessageAlert:
```
return <MessageAlert message={isReleasesError.info.message} severity='error' />
```

Again, it should take precedence over the loading state so should be moved above `if (isReleasesLoading) { ...` to prevent any bugs down the line (if someone changes the conditions of the loading state)